[icu] Build error installing icu[core]:x64-linux -> 70.1

**Host Environment**
- Ubuntu 20.04.4 LTS
- Compiler: gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install icu:x64-linux`

The output will be:
```
./vcpkg install icu:x64-linux                                                              
Computing installation plan...
The following packages will be built and installed:
    icu[core]:x64-linux -> 70.1
Detecting compiler hash for triplet x64-linux...
Restored 0 packages from [...]/.cache/vcpkg/archives in 5.522 us. Use --debug to see more details.
Starting package 1/1: icu:x64-linux
Building package icu[core]:x64-linux...
-- Using cached icu4c-70_1-src.tgz.
-- Cleaning sources at [...]/vcpkg/buildtrees/icu/src/c-70_1-src-a6ee276eef.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source [...]/vcpkg/downloads/icu4c-70_1-src.tgz
-- Applying patch disable-escapestr-tool.patch
-- Applying patch remove-MD-from-configure.patch
-- Applying patch fix_parallel_build_on_windows.patch
-- Applying patch fix-extra.patch
-- Applying patch mingw-dll-install.patch
-- Applying patch disable-static-prefix.patch
-- Applying patch fix-win-build.patch
-- Using source at [...]/vcpkg/buildtrees/icu/src/c-70_1-src-a6ee276eef.clean
-- Getting CMake variables for x64-linux-dbg
-- Getting CMake variables for x64-linux-rel
-- Generating configure for x64-linux
-- Finished generating configure for x64-linux
-- Configuring x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:128 (message):
    Command failed: /bin/bash -c "V=1 ./../src/c-70_1-src-a6ee276eef.clean/source/configure  \"--disable-samples\" \"--disable-tests\" \"--disable-layoutex\" \"--disable-silent-rules\" \"--verbose\" \"--disable-shared\" \"--enable-static\" \"--enable-debug\" \"--disable-release\" \"--prefix=[...]/vcpkg/installed/x64-linux/debug\" \"--bindir=\\${prefix}/../tools/icu/debug/bin\" \"--sbindir=\\${prefix}/../tools/icu/debug/sbin\" \"--libdir=\\${prefix}/lib\" \"--includedir=\\${prefix}/../include\" \"--datarootdir=\\${prefix}/share/icu\""
    Working Directory: [...]/vcpkg/buildtrees/icu/x64-linux-dbg
    Error code: 2
    See logs for more information:
      [...]/vcpkg/buildtrees/icu/config-x64-linux-dbg-out.log
      [...]/vcpkg/buildtrees/icu/config-x64-linux-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_make.cmake:815 (vcpkg_execute_required_process)
  ports/icu/portfile.cmake:52 (vcpkg_configure_make)
  scripts/ports.cmake:145 (include)


error: building icu:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `./vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+icu
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[icu]+Build+error
including:
  package: icu[core]:x64-linux -> 70.1
    vcpkg-tool version: 2022-03-30-692785ac944e81417840c6de244fb3e18a4b35eb
    vcpkg-scripts version: 2ac61f87f 2022-04-22 (27 hours ago)
```

**Failure logs**
vcpkg/buildtrees/icu/config-x64-linux-dbg-err.log:
```
configure: WARNING: unrecognized options: --disable-silent-rules
./../src/c-70_1-src-a6ee276eef.clean/source/configure: line 7981: syntax error near unexpected token 'newline'
./../src/c-70_1-src-a6ee276eef.clean/source/configure: line 7981: 'AX_CHECK_COMPILE_FLAG('
```

**Additional context**
This issue is relevant since it prevents building qt5-base:x64-linux which depends on icu.

--- a/ports/icu/check-autoconf-archive.patch
+++ b/ports/icu/check-autoconf-archive.patch
@@ -1,0 +1,14 @@
+diff --git a/source/configure.ac b/source/configure.ac
+index d194c6d..33fcb2f 100644
+--- a/source/configure.ac
++++ b/source/configure.ac
+@@ -208,6 +208,8 @@ fi
+ #AC_CHECK_PROG(AUTOCONF, autoconf, autoconf, true)
+ #AC_CHECK_PROG(STRIP, strip, strip, true)
+ 
++m4_ifndef([AX_CHECK_COMPILE_FLAG], [AC_MSG_ERROR(['autoconf-archive' is missing])])
++
+ # TODO(ICU-20301): Remove fallback to Python 2.
+ AC_CHECK_PROGS(PYTHON, python3 "py -3" python "py")
+ AC_SUBST(PYTHON)
+

--- a/ports/icu/portfile.cmake
+++ b/ports/icu/portfile.cmake
@@ -1,4 +1,4 @@
-set(ICU_VERSION_MAJOR 70)
+set(ICU_VERSION_MAJOR 71)
 set(ICU_VERSION_MINOR 1)
 set(VERSION "${ICU_VERSION_MAJOR}.${ICU_VERSION_MINOR}")
 set(VERSION2 "${ICU_VERSION_MAJOR}_${ICU_VERSION_MINOR}")
@@ -8,7 +8,7 @@ vcpkg_download_distfile(
     ARCHIVE
     URLS "https://github.com/unicode-org/icu/releases/download/release-${VERSION3}/icu4c-${VERSION2}-src.tgz"
     FILENAME "icu4c-${VERSION2}-src.tgz"
-    SHA512 0b26ae7207155cb65a8fdb25f7b2fa4431e74b12bccbed0884a17feaae3c96833d12451064dd152197fd6ea5fd3adfd95594284a463e66c82e0d860f645880c9
+    SHA512 1fd2a20aef48369d1f06e2bb74584877b8ad0eb529320b976264ec2db87420bae242715795f372dbc513ea80047bc49077a064e78205cd5e8b33d746fd2a2912
 )
 
 vcpkg_extract_source_archive_ex(

--- a/ports/icu/portfile.cmake
+++ b/ports/icu/portfile.cmake
@@ -22,6 +22,7 @@ vcpkg_extract_source_archive_ex(
         mingw-dll-install.patch
         disable-static-prefix.patch # https://gitlab.kitware.com/cmake/cmake/-/issues/16617; also mingw.
         fix-win-build.patch
+        check-autoconf-archive.patch
 )
 
 vcpkg_find_acquire_program(PYTHON3)

--- a/ports/icu/vcpkg.json
+++ b/ports/icu/vcpkg.json
@@ -1,6 +1,6 @@
 {
   "name": "icu",
-  "version": "70.1",
+  "version": "71.1",
   "description": "Mature and widely used Unicode and localization library.",
   "homepage": "https://icu.unicode.org/home",
   "supports": "!uwp",

--- a/ports/icu/vcpkg.json
+++ b/ports/icu/vcpkg.json
@@ -3,6 +3,7 @@
   "version": "71.1",
   "description": "Mature and widely used Unicode and localization library.",
   "homepage": "https://icu.unicode.org/home",
+  "license": "ICU",
   "supports": "!uwp",
   "dependencies": [
     {

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2817,7 +2817,7 @@
       "port-version": 0
     },
     "icu": {
-      "baseline": "70.1",
+      "baseline": "71.1",
       "port-version": 0
     },
     "ideviceinstaller": {

--- a/versions/i-/icu.json
+++ b/versions/i-/icu.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "13fe0c9056059654ea7e482cd06ff70c26f9e029",
+      "version": "71.1",
+      "port-version": 0
+    },
+    {
       "git-tree": "2d082108b0899697c9b6253c260188728374e56b",
       "version": "70.1",
       "port-version": 0

